tools/automatic_query_fixer/src/test/java/com/google/cloud/bigquery/utils/queryfixer/BigQueryServiceTest.java

```diff
@@ -0,0 +1,67 @@
+package com.google.cloud.bigquery.utils.queryfixer;
+
+import com.google.cloud.bigquery.BigQueryException;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.FieldList;
+import com.google.cloud.bigquery.Job;
+import com.google.cloud.bigquery.JobStatistics;
+import com.google.cloud.bigquery.utils.queryfixer.service.BigQueryService;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.collection.IsIterableContainingInOrder.contains;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class BigQueryServiceTest {
+
+  private BigQueryService service;
+
+  /**
```

Comment:
What does it mean here? Where do we read in the credentials?

Reply:
This method requires connection with BigQuery server. 

Uses can call `gcloud auth application-default login` and the credential will stored in the default path, which will be read by this method.

However, uses can download a service account credential and put it elsewhere. If this is the case, they need to modify the test cases manually.

I will put this demonstration in the comment

Reply:
It is also possible to mock the BigQuery server, but I am not sure how long it will take. I guess connecting to real BigQuery and uses the public datasets may be an convenient way to test?